lib/Cmts.ml

```diff
@@ -518,20 +518,37 @@ module Unwrapped = struct
     String.drop_prefix first_line fl_spaces
     :: List.map ~f:(fun s -> String.drop_prefix s min_indent) tl_lines
 
-  let fmt_multiline_cmt ?epi ~opn_pos ~starts_with_sp first_line tl_lines =
+  let fmt_multiline_cmt ?epi ~opn_pos ~starts_with_sp first_line tl_lines
+      ~preserve_indent =
     let open Fmt in
     let is_white_line s = String.for_all s ~f:Char.is_whitespace in
     let unindented = unindent_lines ~opn_pos first_line tl_lines in
-    let fmt_line ~first ~last:_ s =
+    let unindented =
+      if preserve_indent then List.map unindented ~f:(fun x -> (`In_v, x))
+      else
+        List.fold_left unindented ~init:(`Out_v, [])
+          ~f:(fun (prev_state, lines) line ->
+            let next_state =
+              if String.is_prefix line ~prefix:"{v" then `In_v
```

Comment:
This parsing might be a bit weak. One failure state I can think of is the closing not being recognized because there's some whitespace after the `v}`. This would misindent the whole comment.
An other failure is that `{v a single line v}` might trigger the `In_v` state instead of `Out_v`.

Reply:
I fixed these cases.